domain/secret/service/exportimport.go

```diff
@@ -140,7 +140,130 @@ func (s *SecretService) GetSecretsForExport(ctx context.Context) (*SecretExport,
 }
 
 // ImportSecrets saves the supplied secret details to the model.
-func (s *SecretService) ImportSecrets(context.Context, *SecretExport) error {
-	// TODO(secrets)
+func (s *SecretService) ImportSecrets(ctx context.Context, modelSecrets *SecretExport) error {
+	if err := s.importRemoteSecrets(ctx, modelSecrets.RemoteSecrets); err != nil {
+		return errors.Annotate(err, "importing remote secrets")
+	}
+
+	for _, md := range modelSecrets.Secrets {
+		revisions := modelSecrets.Revisions[md.URI.ID]
+		content := modelSecrets.Content[md.URI.ID]
+		if err := s.importSecretRevisions(ctx, md, revisions, content); err != nil {
+			return errors.Annotatef(err, "saving secret %q", md.URI.ID)
+		}
+		for _, sc := range modelSecrets.Consumers[md.URI.ID] {
+			if err := s.st.SaveSecretConsumer(ctx, md.URI, sc.Accessor.ID, &coresecrets.SecretConsumerMetadata{
+				Label:           sc.Label,
+				CurrentRevision: sc.CurrentRevision,
+			}); err != nil {
+				return errors.Annotatef(err, "saving secret consumer %q for %q", sc.Accessor.ID, md.URI.ID)
+			}
+		}
+
+		for _, rc := range modelSecrets.RemoteConsumers[md.URI.ID] {
+			if err := s.st.SaveSecretRemoteConsumer(ctx, md.URI, rc.Accessor.ID, &coresecrets.SecretConsumerMetadata{
+				Label:           rc.Label,
+				CurrentRevision: rc.CurrentRevision,
+			}); err != nil {
+				return errors.Annotatef(err, "saving secret remote consumer %q for %q", rc.Accessor.ID, md.URI.ID)
+			}
+		}
+
+		for _, access := range modelSecrets.Access[md.URI.ID] {
+			p := grantParams(SecretAccessParams{
+				Scope: SecretAccessScope{
+					Kind: access.Scope.Kind,
+					ID:   access.Scope.ID,
+				},
+				Subject: SecretAccessor{
+					Kind: access.Subject.Kind,
+					ID:   access.Subject.ID,
+				},
+				Role: access.Role,
+			})
+			if err := s.st.GrantAccess(ctx, md.URI, p); err != nil {
+				return errors.Annotatef(err, "saving secret access for %q", md.URI.ID)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *SecretService) importSecretRevisions(
+	ctx context.Context, md *coresecrets.SecretMetadata,
+	revisions []*coresecrets.SecretRevisionMetadata,
+	content map[int]coresecrets.SecretData,
+) error {
+	for i, rev := range revisions {
+		params := secret.UpsertSecretParams{
+			ValueRef: rev.ValueRef,
+		}
+		if rev.ValueRef == nil {
+			if data, ok := content[rev.Revision]; ok {
+				params.Data = data
+			}
+		}
+		if i == len(revisions)-1 {
+			params.ExpireTime = md.LatestExpireTime
+		}
+		if i == 0 {
```

Comment:
If it's the 1st revision, we use `CreateSecret`, other revisions use `UpdateSecret`?

Reply:
yup